sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -263,97 +330,117 @@ class kernel_bundle_impl
-                              KernelNames.begin(), KernelNames.end(),
-                              std::inserter(Conflicts, Conflicts.begin()));
-        SeenKernelNames.insert(KernelNames.begin(), KernelNames.end());
-      }
+        DevImages.reserve(DevImages.size() + DevImagesSet.size());
+        for (auto It = DevImagesSet.begin(); It != DevImagesSet.end();)
+          DevImages.push_back(createSyclObjFromImpl<device_image_plain>(
+              *DevImagesSet.extract(It++).value()));
 
-      if (!Conflicts.empty()) {
-        std::stringstream MsgS;
-        MsgS << "Conflicting kernel definitions: ";
-        for (const std::string_view &Conflict : Conflicts)
-          MsgS << " " << Conflict;
-        throw sycl::exception(make_error_code(errc::invalid), MsgS.str());
-      }
-    }
+        // Check for conflicting kernels in RTC kernel bundles.
+        ThrowIfConflictingKernels(DevImages);
 
-    // Create a map between exported symbols and their indices in the device
-    // images collection.
-    std::map<std::string_view, size_t> ExportMap;
-    for (size_t I = 0; I < DevImages.size(); ++I) {
-      device_image_impl &DevImageImpl = *getSyclObjImpl(DevImages[I]);
-      if (DevImageImpl.get_bin_image_ref() == nullptr)
-        continue;
-      for (const sycl_device_binary_property &ESProp :
-           DevImageImpl.get_bin_image_ref()->getExportedSymbols()) {
-        if (ExportMap.find(ESProp->Name) != ExportMap.end())
-          throw sycl::exception(make_error_code(errc::invalid),
-                                "Duplicate exported symbol \"" +
-                                    std::string{ESProp->Name} +
-                                    "\" found in binaries.");
-        ExportMap.emplace(ESProp->Name, I);
+        // Create and insert the corresponding link graph.
+        DevImageLinkGraphs.emplace(&Dev, CreateLinkGraph(DevImages));
       }
-    }
-
-    // Create dependency mappings.
-    std::vector<std::vector<size_t>> Dependencies;
-    Dependencies.resize(DevImages.size());
-    for (size_t I = 0; I < DevImages.size(); ++I) {
-      device_image_impl &DevImageImpl = *getSyclObjImpl(DevImages[I]);
-      if (DevImageImpl.get_bin_image_ref() == nullptr)
-        continue;
-      std::set<size_t> DeviceImageDepsSet;
-      for (const sycl_device_binary_property &ISProp :
-           DevImageImpl.get_bin_image_ref()->getImportedSymbols()) {
-        auto ExportSymbolIt = ExportMap.find(ISProp->Name);
-        if (ExportSymbolIt == ExportMap.end())
-          throw sycl::exception(make_error_code(errc::invalid),
-                                "No exported symbol \"" +
-                                    std::string{ISProp->Name} +
-                                    "\" found in linked images.");
-        DeviceImageDepsSet.emplace(ExportSymbolIt->second);
+    } else {
+      // Collect all unique images.
+      std::vector<device_image_plain> DevImages;
```

Comment:
Can you confirm that `else` part is just indentation change?

Reply:
That is indeed the intention! 😄